source/mqtt/MqttClient.cpp

```diff
@@ -104,6 +104,7 @@ namespace Aws
                 MqttConnection *connection;
                 OnOperationCompleteHandler onOperationComplete;
                 const char *topic;
+                ByteBuf *payload;
```

Comment:
This should not be a pointer type. Then it needs to be initialized to an empty/blank buffer `{ allocator = nullptr, len = 0, ptr = nullptr }` so a call to cleanup if it's still in that state is still a safe no-op.